R/MsBackendMzR.R

```diff
@@ -88,9 +88,14 @@ setMethod("readMsObject", signature(object = "MsBackendMzR",
 #'
 #' @rdname AlabasterParam
 setMethod("saveObject", "MsBackendMzR", function(x, path, ...) {
+    .save_object_spectra_data(x, path, object = "ms_backend_mz_r")
+})
+
+.save_object_spectra_data <- function(x, path, object, version = "1.0") {
     dir.create(path = path, recursive = TRUE, showWarnings = FALSE)
-    saveObjectFile(path, "ms_backend_mz_r",
-                   list(ms_backend_mz_r =list(version = "1.0")))
+    l <- list(list(version = version))
```

Comment:
I am a bit confused about the `list(list())`. and then you name it 'ms_backend_mzr` into it next line ?   why does it need to be a list of a list ?

Reply:
This is the particular format `saveObjectFile` want the input parameter, i.e. a `list` with all optional parameters (one of them being `version` and the name of that `list` should be the name/type of the object. It's confusing, I agree, but I don't know how to call it differently, given that we want to support setting the type with parameter `object` of the upstream function.